x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java

```diff
@@ -274,7 +273,9 @@ public static Query doTranslate(BinaryComparison bc, TranslatorHandler handler)
             return handler.wrapFunctionQuery(bc, bc.left(), () -> translate(bc, handler));
         }
 
-        private static Query translate(BinaryComparison bc, TranslatorHandler handler) {
+        static Query translate(BinaryComparison bc, TranslatorHandler handler) {
+            assert bc.left() instanceof FieldAttribute;
```

Comment:
`assert` is very bad since in the very unlikely case that it fails, it brings the whole JVM down.
Throw an exception instead (see `Check` utility class).